tests/admin/controller/test_individual_admins.py

```diff
@@ -0,0 +1,415 @@
+        test_changing_password(manager2, system)
+        test_changing_password(librarian2, system)
+
+        # Various types of user trying to change a sitewide manager's password
+        test_changing_password(system, sitewide_manager, allowed=True)
+        test_changing_password(sitewide_manager, sitewide_manager, allowed=True)
+        test_changing_password(sitewide_librarian, sitewide_manager)
+        test_changing_password(manager1, sitewide_manager)
+        test_changing_password(librarian1, sitewide_manager)
+        test_changing_password(manager2, sitewide_manager)
+        test_changing_password(librarian2, sitewide_manager)
+
+        # Various types of user trying to change a sitewide librarian's password
+        test_changing_password(system, sitewide_librarian, allowed=True)
+        test_changing_password(sitewide_manager, sitewide_librarian, allowed=True)
+        test_changing_password(manager1, sitewide_librarian, allowed=True)
+        test_changing_password(manager2, sitewide_librarian, allowed=True)
+        test_changing_password(sitewide_librarian, sitewide_librarian)
+        test_changing_password(librarian1, sitewide_librarian)
+        test_changing_password(librarian2, sitewide_librarian)
+
+        # Various types of user trying to change a manager's password
+        # Manager 1
+        test_changing_password(system, manager1, allowed=True)
+        test_changing_password(sitewide_manager, manager1, allowed=True)
+        test_changing_password(manager1, manager1, allowed=True)
+        test_changing_password(sitewide_librarian, manager1)
+        test_changing_password(manager2, manager1)
+        test_changing_password(librarian2, manager1)
+        # Manager 2
+        test_changing_password(system, manager2, allowed=True)
+        test_changing_password(sitewide_manager, manager2, allowed=True)
+        test_changing_password(manager2, manager2, allowed=True)
+        test_changing_password(sitewide_librarian, manager2)
+        test_changing_password(manager1, manager2)
+        test_changing_password(librarian1, manager2)
+
+        # Various types of user trying to change a librarian's password
+        # Librarian 1
+        test_changing_password(system, librarian1, allowed=True)
+        test_changing_password(sitewide_manager, librarian1, allowed=True)
+        test_changing_password(manager1, librarian1, allowed=True)
+        test_changing_password(sitewide_librarian, librarian1)
+        test_changing_password(manager2, librarian1)
+        test_changing_password(librarian2, librarian1)
+        # Librarian 2
+        test_changing_password(system, librarian2, allowed=True)
+        test_changing_password(sitewide_manager, librarian2, allowed=True)
+        test_changing_password(manager2, librarian2, allowed=True)
+        test_changing_password(sitewide_librarian, librarian2)
+        test_changing_password(manager1, librarian2)
+        test_changing_password(librarian1, librarian2)
```

Comment:
This wasn't in the code before, but it's now obvious that we don't test that a librarian can change their own password, and we should add that test.